### PR TITLE
fix: Fix CircleCI integration tests

### PR DESCRIPTION
**Pull request by jamesaucode**
#### Description of changes
This will unblock our code pipeline. The `integ_rn_ios_push_notifications` and `integ_rn_android_storage` integration tests were failing.

- Re-enabled the Android multipart upload progress tests
- Removed the `Install node@10` step as it's no longer needed and is breaking some of the RN tests
- Upgraded the executor of the `integ_rn_ios_push_notifications` test (along with a bump in RN and detox version on the corresponding sample on a separate PR)

#### Issue #, if available
N/A

#### Description of how you validated changes
I created a separate branch on the samples repo and ran CircleCI tests against it (tweaking the `git clone` command to clone the specific branch with the new sample). Here's an example of a successful run:
https://app.circleci.com/pipelines/github/aws-amplify/amplify-js/9968/workflows/c9dd0b9a-625e-440a-b4b1-f8d157367aab

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -257,13 +257,6 @@ commands:
...
           key: |
             brew-cache-{{ arch }}-{{ .Environment.CACHE_VERSION }}
-      - run:
-          # React Native Android can have issues with later versions of Node:
-          # https://github.com/facebook/react-native/issues/26808
-          command: |
-            cd ~/amplify-js
-            ./.circleci/android-rn.sh install-node
-          name: Install node@10
       - run:
```
Comment (hkjpotato) [inline .circleci/config.yml:260]: The issue https://github.com/facebook/react-native/issues/26808 is still open, is it safe to remove this install?

Comment (jamesaucode) [reply .circleci/config.yml:260]: Yes according to this comment: https://github.com/facebook/react-native/issues/26808#issuecomment-642827650

You can also check my manual CircleCI run here: https://app.circleci.com/pipelines/github/aws-amplify/amplify-js/9968/workflows/c9dd0b9a-625e-440a-b4b1-f8d157367aab , the android React Native tests all ran successfully without installing node@10

**Review by iartemiev — APPROVED** [review]
LGTM! Great job, James 🎉

**Review by hkjpotato — APPROVED** [review]
LGTM!

**Pull request merged**